src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +63,990 @@ const ActorInit Dm_Char08_InitVars = {
+    if (ActorCutscene_GetCanPlayNext(nextCs) != 0) {
+        gSaveContext.save.weekEventReg[53] |= 0x40;
+        ActorCutscene_Start(nextCs, &this->dyna.actor);
+        this->actionFunc = func_80AAFE78;
+    } else {
+        ActorCutscene_SetIntentToPlay(nextCs);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF79C.s")
+void func_80AAFAC4(DmChar08* this, PlayState* play) {
+    if (play->csCtx.state == 0) {
+        this->actionFunc = func_80AAF8F4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF884.s")
+void func_80AAFAE4(DmChar08* this, PlayState* play) {
+    if (play->csCtx.state == 0) {
+        this->actionFunc = func_80AAFB04;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF8F4.s")
+void func_80AAFB04(DmChar08* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFA18.s")
+    if ((fabsf(this->dyna.actor.xzDistToPlayer) < 200.0f) && (player->actor.world.pos.y > 0.0f)) {
+        play->nextEntranceIndex = 0x6A70;
+        play->sceneLoadFlag = 0x14;
+        Scene_SetExitFade(play);
+        this->actionFunc = func_80AAFB94;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFAC4.s")
+void func_80AAFB94(DmChar08* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFAE4.s")
+Color_RGBA8 sLargeTurtleBubblePrimColor = { 250, 200, 250, 100 };
+Color_RGBA8 sLargeTurtleBubbleEnvColor = { 80, 80, 80, 0 };
+Vec3f sLargeTurtleBubbelAccel = { 0.0f, -0.8f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFB04.s")
+void DmChar08_SpawnBubbles(DmChar08* this, PlayState* play) {
+    s32 i;
+    Vec3f velocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFB94.s")
+    velocity.x = (Rand_ZeroOne() - 0.5f) * 24.0f;
+    velocity.y = -5.2f;
+    velocity.z = 4.0f;
+    sLargeTurtleBubbelAccel.x = 0.0f;
+    sLargeTurtleBubbelAccel.y = 1.18f;
+    sLargeTurtleBubbelAccel.z = 0.0f;
+    for (i = 0; i < this->bubbleCount; i++) {
+        EffectSsDtBubble_SpawnCustomColor(play, &this->bubblePos, &velocity, &sLargeTurtleBubbelAccel,
+                                          &sLargeTurtleBubblePrimColor, &sLargeTurtleBubbleEnvColor,
+                                          Rand_S16Offset(80, 130), 30, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFBA4.s")
+void func_80AAFCCC(DmChar08* this, PlayState* play) {
+    switch (play->sceneNum) {
+        case SCENE_31MISAKI:
+            if (!(gSaveContext.save.weekEventReg[55] & 0x80)) {
+                switch (this->unk_206) {
+                    case 0:
+                        return;
+                    case 1:
+                        Message_StartTextbox(play, 0x102A, &this->dyna.actor);
```

Comment:
Would be good to leave a small comment above these providing a summary of what these messages say. You can find a spreadsheet of text here: https://docs.google.com/spreadsheets/d/1cbFFzoKaCgqMybAfahhDYPHq1vaCo2pc8v6BfZKhk2M/edit#gid=1312253055